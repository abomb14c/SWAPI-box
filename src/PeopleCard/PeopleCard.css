*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    margin:0 auto;
    text-align: center;
    width: 60%;
}

.person-card {
    background-color: #ffffff;
    border: 2px solid #D6D6D6;
    display: inline-block;
    height: 400px;
    left: 15px;
    margin: 3px;
    position: relative;
    width: 340px;
}

.person-name {
    color: rgba(255,255,255);
    margin: none;
    text-align: center;
    text-shadow: 4px 4px #000;
}

.name-section {
    background-image: url('../assets/peopler-header-section.jpg');
    background-size: cover;
    font-size: 36px;
    height: 125px;
    padding-top: 50px;
}

.content-div{
    margin: 15px;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}