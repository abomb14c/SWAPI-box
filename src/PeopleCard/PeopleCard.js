import React from 'react';
import './PeopleCard.css';
import propTypes from 'prop-types';

const PeopleCard = (props) => {
  const{name, planet, species, population, findFavoritePerson, id, category} = props;
 
  return (
    <article className="person-card">
      <section className="name-section">
        <h1 className="person-name">{name}</h1>
        <button onClick={()=>findFavoritePerson(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Origin</p>
        <h3 className="description-content">{planet}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Species</p>
        <h3 className="description-content">{species}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Population</p>
        <h3 className="description-content">{population}</h3>
      </div>
    </article>
  );
};

PeopleCard.propTypes = {
  name: propTypes.string,
  planet: propTypes.string,
  species: propTypes.string,
  population: propTypes.number,
  findFavoritePerson: propTypes.func,
  id: propTypes.string,
  category: propTypes.string
};
export default PeopleCard;