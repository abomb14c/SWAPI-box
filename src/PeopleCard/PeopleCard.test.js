import React from 'react';
import ReactDOM from 'react-dom';
import PeopleCard from './PeopleCard';
import {shallow, mount} from 'enzyme';
it('should match the snapshot', () => {
  let name='Alan';
  let planet= 'Earth';
  let species='humanish';
  let population= 1;
  let findFavorite= jest.fn();
  let id = "Person0";
  let category= "people";

  let wrapper= shallow(<PeopleCard name={name}
    planet={planet}
    species={species}
    population={population}
    findFavoritePerson={findFavorite}
    id={id}
    category={category}
  />);
  expect(wrapper).toMatchSnapshot();
});