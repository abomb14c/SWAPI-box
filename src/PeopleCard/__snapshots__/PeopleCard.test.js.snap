// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeopleCard
    category="people"
    findFavoritePerson={[Function]}
    id="Person0"
    name="Alan"
    planet="Earth"
    population={1}
    species="humanish"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="name-section"
>
          <button
                    onClick={[Function]}
          >
                    Favorite
          </button>
          <h1
                    className="person-name"
          >
                    Alan
          </h1>
</section>,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Origin
          </p>
          <h3
                    className="description-content"
          >
                    Earth
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Species
          </p>
          <h3
                    className="description-content"
          >
                    humanish
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Population
          </p>
          <h3
                    className="description-content"
          >
                    1
          </h3>
</div>,
      ],
      "className": "person-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              Favorite
</button>,
            <h1
              className="person-name"
>
              Alan
</h1>,
          ],
          "className": "name-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Alan",
              "className": "person-name",
            },
            "ref": null,
            "rendered": "Alan",
            "type": "h1",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Origin
</p>,
            <h3
              className="description-content"
>
              Earth
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Origin",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Origin",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Earth",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "Earth",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Species
</p>,
            <h3
              className="description-content"
>
              humanish
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Species",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Species",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "humanish",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "humanish",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Population
</p>,
            <h3
              className="description-content"
>
              1
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Population",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Population",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "description-content",
            },
            "ref": null,
            "rendered": 1,
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="name-section"
>
            <button
                        onClick={[Function]}
            >
                        Favorite
            </button>
            <h1
                        className="person-name"
            >
                        Alan
            </h1>
</section>,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Origin
            </p>
            <h3
                        className="description-content"
            >
                        Earth
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Species
            </p>
            <h3
                        className="description-content"
            >
                        humanish
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Population
            </p>
            <h3
                        className="description-content"
            >
                        1
            </h3>
</div>,
        ],
        "className": "person-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                Favorite
</button>,
              <h1
                className="person-name"
>
                Alan
</h1>,
            ],
            "className": "name-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Alan",
                "className": "person-name",
              },
              "ref": null,
              "rendered": "Alan",
              "type": "h1",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Origin
</p>,
              <h3
                className="description-content"
>
                Earth
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Origin",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Origin",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Earth",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "Earth",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Species
</p>,
              <h3
                className="description-content"
>
                humanish
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Species",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Species",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "humanish",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "humanish",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Population
</p>,
              <h3
                className="description-content"
>
                1
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Population",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "description-content",
              },
              "ref": null,
              "rendered": 1,
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
